### PR TITLE
Bug fix scroll-less getMoreItems

### PR DESCRIPTION
**Pull request by yoavva**

### CODE REVIEW

File: packages/gallery/src/components/gallery/proGallery/galleryContainer.js [hunk 1]
```diff
@@ -87,8 +87,40 @@ export class GalleryContainer extends React.Component {
...
+      // entering this clause --> the height changed
+      if (i < 15){
+        this.getMoreItemsIfScrollIsDisabled();
+    }
+  }
+}
+
   componentDidMount() {
-    this.initializeScrollPosition()
+    this.initializeScrollPosition();
```
Comment (yoavva) [inline packages/gallery/src/components/gallery/proGallery/galleryContainer.js:110]: I just added the semicolon on this line

File: packages/gallery/src/components/gallery/proGallery/galleryContainer.js [hunk 2]
```diff
@@ -87,8 +87,40 @@ export class GalleryContainer extends React.Component {
     }
   }
 
+  // This function runs if site is scroll-less and fetches gallery's items
+  async getMoreItemsIfScrollIsDisabled() {
+    const { body, documentElement: html } = document;
+    const viewportHeight = window.innerHeight;
+    const height = Math.max(body.scrollHeight, body.offsetHeight, html.clientHeight, html.scrollHeight, html.offsetHeight );
+    const padding = 100; // Wix Platform adds padding to body height
```
Comment (yhattav) [inline packages/gallery/src/components/gallery/proGallery/galleryContainer.js]: who are Wix?

File: packages/gallery/src/components/gallery/proGallery/galleryContainer.js [hunk 3]
```diff
@@ -87,8 +87,40 @@ export class GalleryContainer extends React.Component {
...
+    
+    //there can be no scroll to trigger getMoreItems, but there could be more items
+    if(height <= viewportHeight + padding) { 
+      // Trying to get more items
+      this.getMoreItemsIfNeeded(0);
+
+      // Loop initiation
+      let nextHeight = Math.max(body.scrollHeight, body.offsetHeight, html.clientHeight, html.scrollHeight, html.offsetHeight);
```
Comment (yhattav) [inline packages/gallery/src/components/gallery/proGallery/galleryContainer.js]: when will const 103 be different than const 94?

File: packages/gallery/src/components/gallery/proGallery/galleryContainer.js [hunk 4]
```diff
@@ -87,8 +87,40 @@ export class GalleryContainer extends React.Component {
...
+      let i = 0
+      while (nextHeight === height && i < 15){
+        // Wait 100 ms for the body to update
+        await new Promise((res) => setTimeout(res, 100));
```
Comment (yhattav) [inline packages/gallery/src/components/gallery/proGallery/galleryContainer.js]: so just a slow while?

**Review by yhattav — CHANGES_REQUESTED** [review]
we said you'll talk to me before writing the PR...

File: packages/gallery/src/components/gallery/proGallery/galleryHelpers.js [hunk 5]
```diff
@@ -1,5 +1,22 @@
 import { window } from 'pro-gallery-lib';
 
+// Promise-wrapping class
+export class Deferred {
+  constructor() {
+    this.promise = new Promise((resolve, reject) => {
+      this.reject = () => {
```
Comment (yhattav) [inline packages/gallery/src/components/gallery/proGallery/galleryHelpers.js]: hey, there should be a fix here. this.reject(...args) =>{  reject(...args)} same with resolve
export class Deferred {
  constructor() {
    this.promise = new Promise((resolve, reject) => {
      this.reject = (...args) => {
        this.isPending = false;
        reject(...args);
      };
      this.resolve = (...args) => {
        this.isPending = false;
        resolve(...args);
      };
      this.isPending = true;
    });
  }
}

**Review by yhattav — CHANGES_REQUESTED** [review]

**Review by yhattav — APPROVED** [review]

**Pull request merged**